extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/DevServicesKafkaProcessor.java

```diff
@@ -0,0 +1,284 @@
+
+        KafkaDevServiceCfg configuration = getConfiguration(kafkaClientBuildTimeConfig);
+
+        if (closeable != null) {
+            boolean shouldShutdownTheBroker = launchMode.getLaunchMode() == LaunchMode.TEST;
+            if (!shouldShutdownTheBroker) {
+                shouldShutdownTheBroker = !configuration.equals(cfg);
+            }
+            if (!shouldShutdownTheBroker) {
+                return null;
+            }
+            shutdownBroker();
+            cfg = null;
+        }
+
+        KafkaBroker kafkaBroker = startKafka(configuration);
+        DevServicesKafkaBrokerBuildItem bootstrapServers = null;
+        if (kafkaBroker != null) {
+            closeable = kafkaBroker.getCloseable();
+            runTimeConfiguration.produce(new RunTimeConfigurationDefaultBuildItem(
+                    KAFKA_BOOTSTRAP_SERVERS, kafkaBroker.getBootstrapServers()));
+            bootstrapServers = new DevServicesKafkaBrokerBuildItem(kafkaBroker.getBootstrapServers());
+        }
+
+        // Configure the watch dog
+        if (first) {
+            first = false;
+            Runnable closeTask = new Runnable() {
+                @Override
+                public void run() {
+                    if (closeable != null) {
+                        shutdownBroker();
+                    }
+                    first = true;
+                    closeable = null;
+                    cfg = null;
+                }
+            };
+            QuarkusClassLoader cl = (QuarkusClassLoader) Thread.currentThread().getContextClassLoader();
+            ((QuarkusClassLoader) cl.parent()).addCloseTask(closeTask);
+            Thread closeHookThread = new Thread(closeTask, "Kafka container shutdown thread");
+            Runtime.getRuntime().addShutdownHook(closeHookThread);
+            ((QuarkusClassLoader) cl.parent()).addCloseTask(new Runnable() {
+                @Override
+                public void run() {
+                    Runtime.getRuntime().removeShutdownHook(closeHookThread);
+                }
+            });
+        }
+        cfg = configuration;
+        return bootstrapServers;
+    }
+
+    private void shutdownBroker() {
+        if (closeable == null) {
+            try {
+                closeable.close();
+            } catch (Throwable e) {
+                log.error("Failed to stop the Kafka broker", e);
+            } finally {
+                closeable = null;
+            }
+        }
+    }
+
+    private KafkaBroker startKafka(KafkaDevServiceCfg config) {
+        if (!config.devServicesEnabled) {
+            // explicitly disabled
+            log.debug("Not starting dev services for Kafka, as it has been disabled in the config.");
+            return null;
+        }
+
+        if (!isDockerWorking.getAsBoolean()) {
+            log.warn("Please configure the Kafka bootstrap servers property (kafka.bootstrap.servers) "
```

Comment:
I'd rephrase this, it's not exactly clear what the problem is. Something like `Docker isn't working, please configure the Kafka bootstrap servers property (kafka.bootstrap.servers)`.